packages/core/src/validators.ts

```diff
@@ -496,7 +490,7 @@ export class ObjectValidator extends Validator {
     };
     const cycleResult = ctx.registerObject(value, path, context.convertedObject);
     if (cycleResult) {
-      return ctx.promise(cycleResult);
+      return ctx.promise(cycleResult as ValidationResult<LocalType>);
```

Comment:
My editor is resolving this as `ValidationResult<any>` so this cast seems unnecessary as well.